### ISSUE TITLE
osxphotos info on macOS 14.6 gives error 'no such table: Z_28ASSETS'

### ISSUE THREAD

**Issue by oPromessa** [RhetTbull/osxphotos#1628]
**Before submitting a bug report, please ensure you are running the most recent version of osxphotos and that the bug is reproducible on the latest version**

I'm doing some tests on Sonoma 14.6 Intel based Mac...

```py
$ sw_vers 
ProductName:		macOS
ProductVersion:		14.6
BuildVersion:		23G80

$ osxphotos --version
osxphotos, version 0.68.2
Python 3.11.4 (main, Jul  5 2023, 09:00:44) [Clang 14.0.6 ]
macOS 10.16.0, x86_64
```

**Describe the bug**
Ran `info` on a newly created Photo library just to check for basic info and got a crash.
Some other functions return similar errors: query, inspect, etc.

**To Reproduce**
1. What' the full command line you used with osxphotos?
2. What was the error output?

```py
$ osxphotos info
Using last opened Photos library: /Users/XXX/Documents/XXX/XXX.photoslibrary
Loading library: /Users/XXX/Documents/XXX/XXX.photoslibrary
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/XXX/Documents/XXX/venv/bin/osxphotos:8 in <module>                                        │
│                                                                                                  │
│   5 from osxphotos.__main__ import cli_main                                                      │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(cli_main())                                                                     │
│   9                                                                                              │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/core.py:1157 in __call__        │
│                                                                                                  │
│   1154 │                                                                                         │
│   1155 │   def __call__(self, *args: t.Any, **kwargs: t.Any) -> t.Any:                           │
│   1156 │   │   """Alias for :meth:`main`."""                                                     │
│ ❱ 1157 │   │   return self.main(*args, **kwargs)                                                 │
│   1158                                                                                           │
│   1159                                                                                           │
│   1160 class Command(BaseCommand):                                                               │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/core.py:1078 in main            │
│                                                                                                  │
│   1075 │   │   try:                                                                              │
│   1076 │   │   │   try:                                                                          │
│   1077 │   │   │   │   with self.make_context(prog_name, args, **extra) as ctx:                  │
│ ❱ 1078 │   │   │   │   │   rv = self.invoke(ctx)                                                 │
│   1079 │   │   │   │   │   if not standalone_mode:                                               │
│   1080 │   │   │   │   │   │   return rv                                                         │
│   1081 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!                               │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/core.py:1688 in invoke          │
│                                                                                                  │
│   1685 │   │   │   │   super().invoke(ctx)                                                       │
│   1686 │   │   │   │   sub_ctx = cmd.make_context(cmd_name, args, parent=ctx)                    │
│   1687 │   │   │   │   with sub_ctx:                                                             │
│ ❱ 1688 │   │   │   │   │   return _process_result(sub_ctx.command.invoke(sub_ctx))               │
│   1689 │   │                                                                                     │
│   1690 │   │   # In chain mode we create the contexts step by step, but after the                │
│   1691 │   │   # base command has been invoked.  Because at that point we do not                 │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/core.py:1434 in invoke          │
│                                                                                                  │
│   1431 │   │   │   echo(style(message, fg="red"), err=True)                                      │
│   1432 │   │                                                                                     │
│   1433 │   │   if self.callback is not None:                                                     │
│ ❱ 1434 │   │   │   return ctx.invoke(self.callback, **ctx.params)                                │
│   1435 │                                                                                         │
│   1436 │   def shell_complete(self, ctx: Context, incomplete: str) -> t.List["CompletionItem"]:  │
│   1437 │   │   """Return a list of completions for the incomplete value. Looks                   │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/core.py:783 in invoke           │
│                                                                                                  │
│    780 │   │                                                                                     │
│    781 │   │   with augment_usage_errors(__self):                                                │
│    782 │   │   │   with ctx:                                                                     │
│ ❱  783 │   │   │   │   return __callback(*args, **kwargs)                                        │
│    784 │                                                                                         │
│    785 │   def forward(                                                                          │
│    786 │   │   __self, __cmd: "Command", *args: t.Any, **kwargs: t.Any  # noqa: B902             │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/decorators.py:45 in new_func    │
│                                                                                                  │
│    42 │   """                                                                                    │
│    43 │                                                                                          │
│    44 │   def new_func(*args: "P.args", **kwargs: "P.kwargs") -> "R":                            │
│ ❱  45 │   │   return f(get_current_context().obj, *args, **kwargs)                               │
│    46 │                                                                                          │
│    47 │   return update_wrapper(new_func, f)                                                     │
│    48                                                                                            │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/click/decorators.py:33 in new_func    │
│                                                                                                  │
│    30 │   """                                                                                    │
│    31 │                                                                                          │
│    32 │   def new_func(*args: "P.args", **kwargs: "P.kwargs") -> "R":                            │
│ ❱  33 │   │   return f(get_current_context(), *args, **kwargs)                                   │
│    34 │                                                                                          │
│    35 │   return update_wrapper(new_func, f)                                                     │
│    36                                                                                            │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/osxphotos/cli/info.py:36 in info      │
│                                                                                                  │
│   33 │                                                                                           │
│   34 │   click.echo(f"Loading library: {db}", err=True)                                          │
│   35 │                                                                                           │
│ ❱ 36 │   photosdb = iPhotoDB(db) if is_iphoto_library(db) else PhotosDB(dbfile=db)               │
│   37 │   info = {"database_path": photosdb.db_path, "database_version": photosdb.db_version}     │
│   38 │   photos = photosdb.photos(movies=False)                                                  │
│   39 │   not_shared_photos = [p for p in photos if not p.shared]                                 │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/osxphotos/photosdb/photosdb.py:419 in │
│ __init__                                                                                         │
│                                                                                                  │
│    416 │   │   if int(self._db_version) <= int(_PHOTOS_4_VERSION):                               │
│    417 │   │   │   self._process_database4()                                                     │
│    418 │   │   else:                                                                             │
│ ❱  419 │   │   │   self._process_database5()                                                     │
│    420 │   │                                                                                     │
│    421 │   │   self._db_connection, _ = self.get_db_connection()                                 │
│    422                                                                                           │
│                                                                                                  │
│ /Users/XXX/Documents/XXX/venv/lib/python3.11/site-packages/osxphotos/photosdb/photosdb.py:1832   │
│ in _process_database5                                                                            │
│                                                                                                  │
│   1829 │   │                                                                                     │
│   1830 │   │   # get details about albums                                                        │
│   1831 │   │   verbose("Processing albums.")                                                     │
│ ❱ 1832 │   │   c.execute(                                                                        │
│   1833 │   │   │   f""" SELECT                                                                   │
│   1834 │   │   │   │   ZGENERICALBUM.ZUUID,                                                      │
│   1835 │   │   │   │   {asset_table}.ZUUID,                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
OperationalError: no such table: Z_28ASSETS
```

**Expected behavior**
Info about the Library.

**Screenshots**
N/A

**Desktop (please complete the following information):**
```py
$ sw_vers 
ProductName:		macOS
ProductVersion:		14.6
BuildVersion:		23G80

$ osxphotos --version
osxphotos, version 0.68.2
Python 3.11.4 (main, Jul  5 2023, 09:00:44) [Clang 14.0.6 ]
macOS 10.16.0, x86_64
```

**Additional context**
N/A

**Comment by oPromessa** [comment]
For reference I attach the schema.log for `Photos.photoslibrary/database/Photos.sqlite`

Table is now named Z_29ASSETS !!

[schema.log](https://github.com/user-attachments/files/16480790/schema.log)

**Comment by RhetTbull** [comment]
Thanks. Already working on this! I will try to get a quick fix out this weekend. I'm also working on a new solution that will allow osxphotos to auto-discover the correct table names as Apple seems to be changing this a lot (it's changed twice already during the macOS 15 beta).

**Comment by RhetTbull** [comment]
Release [v0.68.3](https://github.com/RhetTbull/osxphotos/releases/tag/v0.68.3) should restore functionality for Sequoia beta and macOS 14.6 however I don't have access to these to test. Please let me know if it works or if you encounter any issues.

**Comment by ablyler** [comment]
Thanks @RhetTbull, I upgraded and now I get this error:

`File "/Users/.../.local/share/virtualenvs/osxphotos-pBzELGGF/lib/python3.12/site-packages/osxphotos/photosdb/photosdb.py", line 419, in __init__
    self._process_database5()
  File "/Users/.../.local/share/virtualenvs/osxphotos-pBzELGGF/lib/python3.12/site-packages/osxphotos/photosdb/photosdb.py", line 1951, in _process_database5
    c.execute(
sqlite3.OperationalError: no such column: Z_1KEYWORDS.Z_40KEYWORDS`

**Comment by RhetTbull** [comment]
Ok will take a look. I don't have access to my 14.x machine at the moment so doing all of this blind.

**Comment by ablyler** [comment]
> Ok will take a look. I don't have access to my 14.x machine at the moment so doing all of this blind.

I'm happy to help test or provide anything that would help make this easier for you.

**Comment by ablyler** [comment]
@RhetTbull I was able to fix this locally and created a PR: https://github.com/RhetTbull/osxphotos/pull/1632

**Comment by RhetTbull** [comment]
Thanks!

**Comment by RhetTbull** [comment]
I think this is finally fixed in [v0.68.4](https://github.com/RhetTbull/osxphotos/releases/tag/v0.68.4). Thanks @ablyler and @oPromessa for providing test libraries which are now incorporated into the test suite!

**Comment by maxexcloo** [comment]
Really appreciate the fix everyone 🙏🏻
